apollo-compiler/src/main/kotlin/com/apollographql/apollo/compiler/backend/ir/BackendIrBuilder.kt

```diff
@@ -52,6 +54,42 @@ internal class BackendIrBuilder constructor(
     )
   }
 
+  /**
+   * a GroupedField is a list of fields with the same responseName and arguments but possibly different selectionSets and directives
+   */
+  private data class GroupedField(val fields: List<GQLField>)
+
+  private fun GQLValue.toBoolean(): Boolean? = when(this) {
+    is GQLBooleanValue -> this.value
+    else -> null
+  }
+
+  private fun GQLDirective.toBoolean(): Boolean? {
+    return when (name) {
+      "include" -> arguments!!.arguments.first().value.toBoolean()
+      "skip" -> arguments!!.arguments.first().value.toBoolean()?.not()
+      else -> null
+    }
+  }
+  private fun GroupedField.removeLiteralDirectives(): GroupedField? {
+    val newFields = fields.mapNotNull {
+      val isAlwaysSkipped = it.directives.firstOrNull { it.toBoolean() == false } != null
+      if (isAlwaysSkipped) {
+        // 3.13.2 the field or fragment must not be queried if either the @skip condition is true or the @include condition is false.
+        null
+      } else {
+        // Directives that are always true don't add any useful information
+        it.copy(directives = it.directives.filter {
+          it.toBoolean() != true
+        })
+      }
+    }
+    if (newFields.isEmpty()) {
+      return null
+    } else {
+      return copy(fields = newFields)
+    }
+  }
```

Comment:
Can we move this logic out of the `BackendIrBuilder.kt` into separate file? We do have `FrontendIrMergeUtils`, do you think it will be better to move this logic over there?

Reply:
This logic is about eliminating literal `@include`/`@skip` directives. I can move them to a separate `IncludeSkipUtils` file but it's not really about merging so I wouldn't use `FrontendIrMergeUtils`

Reply:
Yeah I would suggest to move it out from the builder as it keep growing fast, so might be good `IncludeSkipUtils `.